pandas/io/parsers.py

```diff
@@ -913,8 +917,21 @@ def read(self, nrows=None):
         # May alter columns / col_dict
         index, columns, col_dict = self._create_index(ret)
 
+        if index is None:
+            if col_dict:
+                # Any column is actually fine:
+                new_rows = len(compat.next(compat.itervalues(col_dict)))
+                index = RangeIndex(self._currow, self._currow + new_rows)
+            else:
+                new_rows = 0
+                index = Index([])
```

Comment:
An (empty) ``RangeIndex`` here breaks some comparisons with the output of the ``CParser`` with ``low_memory=True``. Both should probably be fixed, but I'm not sure whether we need to first make ``RangeIndex(a, b).append(RangeIndex(b, c))`` return a ``RangeIndex(a, c)`` (rather than an ``Int64Index``, as happens currently).